Packet++/header/TelnetLayer.h

```diff
@@ -0,0 +1,343 @@
+#ifndef PACKETPP_TELNET_LAYER
+#define PACKETPP_TELNET_LAYER
+
+#include "Layer.h"
+
+#include <vector>
+
+/// @file
+
+/**
+ * \namespace pcpp
+ * \brief The main namespace for the PcapPlusPlus lib
+ */
+namespace pcpp
+{
+
+    /**
+     * Class for representing the Telnet Layer
+     */
+    class TelnetLayer : public Layer
+    {
+    private:
+#pragma pack(push, 1)
+        struct telnet_header
+        {
+            // "Interpret as Command" escape character (FF)
+            uint8_t interpretation,
+                // Command
+                command,
+                // Option of the command
+                subcommand,
+                // Data (Variable length)
+                data[];
+        };
+#pragma pack(pop)
+
+        struct telnet_field_data
+        {
+            // Header
+            struct telnet_header *hdr;
+            // Size of the header including data payload
+            uint16_t hdrSize;
+            // Offset of the current header
+            uint16_t currentOffset;
+        };
```

Comment:
1. Generally speaking we usually try to keep as less layer metadata as possible for several reasons:
   1. Avoid taking a lot of memory per packet
   2. In case the current layer of  the layers below it change, there is less data and pointers we need to update 
2. Keeping pointers might be an issue because if the `Packet` object is copied or if one of the other layers changes, these pointers become invalid and can even point to invalid memory. Instead, it's usually better to keep just the offset

You can see examples from other layers, but we usually keep just the number of "options" or "commands" and parse the packet again each time `getOption()` or `getOptionData()` is called